### ISSUE TITLE
Constraints of type `bool`

### ISSUE THREAD

**Issue by adishavit** [cvxgrp/cvxpy#1280]
This falls somewhere between a Feature Request and a Bug Report. In a way it is about philosophy.  

My problem's constraints are frequently built incrementally. 
If some of my variables were initialized with a scalar e.g. `0` instead of `cp.Constant(0)` (as recommended to me [here](#970 )) and these variables are used with some empty loops, then occasionally, in some constraint, I may get a constraint that is tautologically just `True` (or `False`).  
This happens since I would have e.g. `[0 <= 42]` which is not parsed as a CVXPY expression but just a simple, "eager", arithmetic Python condition which results in `[True]`.  

At the moment, CVXPY crashes with these constraints, and I have to either replace all scalar initializers with eg `cp.Constant(0)` so we get valid expressions or check and remove these manually.

I would expect that:

- If the list of problem constraints contain pure `True` entries, they may be safely and silently ignored.
- If the list of problem constraints contain pure `False` entries, then the problem is by definition infeasible - so there is no use trying to solve it.

IMO in both cases crashing is the worst option.

As a temporary measure, is it legal to modify `Problem._constaints` to remove such `bool` `True` entries (to avoid crashes) _after_ the `Problem` is created?

**Comment by SteveDiamond** [comment]
I agree that cvxpy should have the behavior you describe.

**Comment by adishavit** [comment]
I can add a `True` filter on the input constraints to the `Problem` ctor. 

But how should `False` be handled?  
Raise an exception?  
Is there a way to indicate the problem is infeasible already at the ctor stage?

**Comment by SteveDiamond** [comment]
Yeah handling False will require more thought. But if you want to do a PR with a True filter that would be at least something. You can't just remove the True's though, because people might depend on the order/index of the constraints. We would need to filter them out in some part of the reduction process.

**Comment by adishavit** [comment]
Good point about the order of the constraints.

How about replacing:

 True with Constant(0) <= Constant(1)

and

False with Constant(1) <= Constant(0)

Wdyt ?

**Comment by SteveDiamond** [comment]
I like it.

**Comment by adishavit** [comment]
Will do 🙂

**Comment by adishavit** [comment]
@SteveDiamond : Seems like you even added [a stub unit test for this](https://github.com/cvxgrp/cvxpy/blob/1b389f050103914a6f629bf47c499fde120c9aff/cvxpy/tests/test_problem.py#L1763) - if I am not mistaken:

```
    def test_bool_constr(self):
        """Test constraints that evaluate to booleans.
        """
        pass
```

If so, I will fill it in and implement as we discussed.

**Comment by SteveDiamond** [comment]
Closed by #1283
